js/piwik.js

```diff
@@ -4266,7 +4266,7 @@ if (typeof window.Piwik !== 'object') {
             function logEvent(category, action, name, value, customData, callback)
             {
                 // Category and Action are required parameters
-                if (trim(String(category)).length === 0 || trim(String(action)).length === 0) {
+                if (typeof category === 'undefined' || typeof action === 'undefined' || trim(String(category)).length === 0 || trim(String(action)).length === 0) {
```

Comment:
You'd probably want to use `isDefined()` or even better maybe `isNumberOrHasLength()`

Reply:
I was not aware that there were helper functions like this.

It took a few tries, but I think I got it right now.